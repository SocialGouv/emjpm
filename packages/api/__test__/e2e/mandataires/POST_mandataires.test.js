//
const request = require("supertest");

const server = require("@emjpm/api/app");

const knex = require("@emjpm/api/db/knex");
const { getTokenByUserType } = require("../utils");

beforeEach(async () => {
  await knex.seed.run();
});

afterAll(async () => {
  await knex.destroy();
});

const defaultRegister = {
  contact_nom: "testMandaAd",
  contact_prenom: "testPrenom",
  etablissement: "",
  contact_email: "totoManda@toto.com",
  type: "service",
  adresse: "",
  code_postal: "75010",
  ville: "",
  telephone: ""
};

const simpler = ({ created_at, ...props }) => props;

test("should register with good values", async () => {
  const token = await getTokenByUserType("service");
  const response = await request(server)
    .post("/api/v1/mandataires")
    .set("Authorization", "Bearer " + token)
    .send(defaultRegister);
  expect(response.body.success).toBe(true);
  expect(response.status).toBe(200);

  const lastInsert = await knex
    .table("mandataires")
    .orderBy("created_at", "desc")
    .first();
  expect(simpler(lastInsert)).toMatchSnapshot();
});
