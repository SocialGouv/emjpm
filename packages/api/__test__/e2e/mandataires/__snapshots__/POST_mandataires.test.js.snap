// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should register with good values 1`] = `
Object {
  "adresse": "",
  "code_postal": "75010",
  "contact_email": "totoManda@toto.com",
  "contact_nom": "testMandaAd",
  "contact_prenom": "testPrenom",
  "date_mesure_update": null,
  "dispo_max": null,
  "email_send": null,
  "etablissement": "",
  "genre": null,
  "id": 6,
  "mesures_en_attente": null,
  "mesures_en_cours": null,
  "nb_secretariat": null,
  "secretariat": null,
  "service_id": null,
  "telephone": "",
  "telephone_portable": null,
  "user_id": 3,
  "ville": "",
  "zip": null,
}
`;
